src/store/tools/useLabels.ts

```diff
@@ -0,0 +1,29 @@
+import { ref } from 'vue';
+
+type LabelProps<Tool> = Partial<Tool>;
+export type Labels<Tool> = Record<string, LabelProps<Tool>>;
+
+export const useLabels = <Tool>(initialLabels: Labels<Tool>) => {
+  const labels = ref(initialLabels);
+
+  const activeLabel = ref<string | undefined>();
+  const setActiveLabel = (name: string) => {
+    activeLabel.value = name;
+  };
+
+  const setLabels = (newLabels: Labels<Tool> | null | undefined) => {
```

Comment:
FYI don't forget `Maybe<Labels<Tool>>` will simplify `| undefined` and `| null | undefined` for cases where we don't care about null/undefined distinction.

Reply:
Oh, yeah!